### PR TITLE
Bot Answers

### PR DESCRIPTION
**Pull request by JamesAdamHughes**
Players can select to play with up to 6 bots during the game. They submit answers to questions but cannot vote. 
Users can pick those answers, which are awarded to bots and persist throughout the game. 
They are also included in the ranking system.

Also put user and room functions into the main body and used the self keyword, not the this

### CODE REVIEW

**Comment by JamesAdamHughes** [comment]
bloody circleCi is having a nightmare

**Comment by JamesAdamHughes** [comment]
Going on the release after this

**Comment by JamesAdamHughes** [comment]
DONT PULL YET

**Comment by JamesAdamHughes** [comment]
Ready for review @BRHoward @Geor23

File: server/server.js [hunk 1]
```diff
@@ -449,7 +452,8 @@ module.exports = function(port, enableLogging) {
 
         function putUserInJoining() {
             socket.emit("ROUTING", {
-                location: 'joining'
+                location: 'joining',
+                messgae: "fuck"
```
Comment (BRHoward) [inline server/server.js]: inappropriate

**Comment by JamesAdamHughes** [comment]
Also fixed isue #106

**Comment by BRHoward** [comment]
looks good

File: client/includes/templates/game/vote_favourite.html [hunk 2]
```diff
@@ -12,11 +12,11 @@ <h2 ng-init="startCountdown()">{{roundQuestion()}}</h2>
             <div class="col-md-12 text-center" ng-repeat="answer in answers() | orderBy:'answerText'">
                 <div class="card-text-container">
                     <button ng-if="getUserId() === answer.player.uId" class="btn btn-warning btn-lg btn-block card-text-button" type="button" id="ownAnswer" ng-disabled="true"> {{answer.answerText}} <i>(Your answer)</i></button>
-                    <button ng-if="getUserId() !== answer.player.uId" class="btn btn-warning btn-lg btn-block card-text-button" type="button" id="answer" ng-click="submitVote(answer.answerText)">{{answer.answerText}}</button>
+                    <button ng-if="getUserId() !== answer.player.uId" class="btn btn-warning btn-lg btn-block card-text-button" type="button" id="answer" ng-click="submitVote(answer)">{{answer.answerText}}</button>
                     </button>
```
Comment (Geor23) [inline client/includes/templates/game/vote_favourite.html]: extra "</button>" tag

File: server/GameController.js [hunk 3]
```diff
@@ -225,14 +220,23 @@ module.exports = function(data) {
...
 
-			//check if everyone submitted and sends back all the currently submitted answers
 			var allChoicesSubmitted;
 
+			//check if everyone submitted and sends back all the currently submitted answers
 			if (currentRound.answers.length === getNumOfConnectedPlayers()) {
 				// change gametsate to the next stage
+
+				//allow everyone to vote again
+				setAllPlayersAbleToSubmit();
```
Comment (Geor23) [inline server/GameController.js]: Don't need this as the function setAllPlayersAbleToSubmit() is also called on page load -> aka when timer is started

File: server/GameController.js [hunk 4]
```diff
@@ -215,7 +210,7 @@ module.exports = function(data) {
 			var ans = {
 				player: submittingPlayer,
 				answerText: answerText,
-				playersVote: []
+				playersVote: [],
```
Comment (Geor23) [inline server/GameController.js]: no need for ','

**Pull request merged**